ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -764,6 +786,40 @@ private static ConfigDef buildConfigDef(final ConfigGeneration generation) {
                 + "will prevent connect from starting up embedded within KSQL. For more information"
                 + " on configuring connect, see "
                 + "https://docs.confluent.io/current/connect/userguide.html#configuring-workers."
+        ).define(
+            CONNECT_BASIC_AUTH_CREDENTIALS_SOURCE_PROPERTY,
+            ConfigDef.Type.STRING,
+            BASIC_AUTH_CREDENTIALS_SOURCE_NONE,
+            BASIC_AUTH_CREDENTIALS_SOURCE_VALIDATOR,
+            Importance.LOW,
+            "If providing explicit basic auth credentials for ksqlDB to use when sending connector "
+                + "requests, this config specifies how credentials should be loaded. Valid "
+                + "options are 'FILE' in order to specify the username and password in a "
+                + "properties file, or 'NONE' to indicate that custom basic auth should "
+                + "not be used. If 'NONE', ksqlDB will forward the auth header, if present, "
+                + "on the incoming ksql request to Connect."
+        ).define(
+            CONNECT_BASIC_AUTH_CREDENTIALS_FILE_PROPERTY,
+            ConfigDef.Type.STRING,
+            "",
+            Importance.LOW,
+            "If '" + CONNECT_BASIC_AUTH_CREDENTIALS_SOURCE_PROPERTY + "' is set to 'FILE', "
+                + "then this config specifies the path to the credentials file."
+        ).define(
+            CONNECT_BASIC_AUTH_CREDENTIALS_RELOAD_PROPERTY,
+            ConfigDef.Type.BOOLEAN,
+            false,
+            ConfigDef.Importance.LOW,
+            "If true, basic auth credentials for connector auth will automatically reload "
+                + "on file change (creation or modification). File deletion is not monitored and "
+                + "old credentials will continue to be used in this case."
+        ).define(
+            CONNECT_BASIC_AUTH_FAIL_ON_UNREADABLE_CREDENTIALS,
+            ConfigDef.Type.BOOLEAN,
+            true,
+            ConfigDef.Importance.LOW,
+            "If true, failure to load basic auth credentials for connector auth will result "
+                + "in an error. If false, failure will result in a warn log and empty credentials."
```

Comment:
Why do we need this failure optional? If an error occurs with Connect, then it's a good warning to the user that Connect/KSQL settings are not well configured, right? 

Cloud users don't see the logs, just the client errors.

Reply:
Discussed offline -- we're going to remove this config and default the behavior to false, i.e., failure to load credentials will not throw an error from the engine. Instead, the error will be logged to the server log and the user will receive a generic 401 from the failed request, rather than a message indicating that the credentials file could be read. We think this behavior is fine from the user's POV since they can access the server logs to view the error, and we generally want to minimize the number of server configs introduced.

I'm going to make these changes in my follow-up PR (https://github.com/confluentinc/ksql/pull/8357) in order to merge this one sooner to unblock the other one.